zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -0,0 +1,382 @@
+
+    def test_102_cold_start_bootstrap(self):
+        """Bootstrap a non-leader node.
+
+        After bootstrapping a non-leader node, notify bootstrapped on the
+        leader node.
+        """
+        # Update which node is the leader and which are not
+        self.update_leaders_and_non_leaders()
+        # We want to test the worst possible scenario which is the
+        # non-leader with the highest sequence number. We will use the leader
+        # for the notify-bootstrapped after. They just need to be different
+        # units.
+        logging.info("Execute bootstrap-pxc action after cold boot ...")
+        zaza.model.run_action(
+            self.non_leaders[0],
+            "bootstrap-pxc",
+            action_params={})
+        logging.debug("Wait for application states ...")
+        states = {"percona-cluster": {
+            "workload-status": "waiting",
+            "workload-status-message": "Unit waiting for cluster bootstrap"}}
+        zaza.model.wait_for_application_states(
+            states=states)
+        logging.info("Execute notify-bootstrapped action after cold boot on "
+                     "the leader node ...")
+        zaza.model.run_action_on_leader(
+            self.application,
+            "notify-bootstrapped",
+            action_params={})
+        logging.debug("Wait for application states ...")
+        test_config = lifecycle_utils.get_charm_config()
+        zaza.model.wait_for_application_states(
+            states=test_config.get("target_deploy_status", {}))
+
+
+class PerconaClusterScaleTests(PerconaClusterTest):
+    """Percona Cluster scale tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running percona scale tests.
+
+        Note: these have tests have been ported from amulet tests
+        """
+        super(PerconaClusterScaleTests, cls).setUpClass()
+
+    def test_100_kill_crm_master(self):
+        """Ensure VIP failover.
+
+        When killing the mysqld on the crm_master unit verify the VIP fails
+        over.
+        """
+        logging.info("Testing failover of crm_master unit on mysqld failure")
+        # we are going to kill the crm_master
+        old_crm_master = self.get_crm_master()
+        logging.info(
+            "kill -9 mysqld on {}".format(old_crm_master)
+        )
+        cmd = "sudo killall -9 mysqld"
+        zaza.model.run_on_unit(old_crm_master, cmd)
+
+        logging.info("looking for the new crm_master")
+        i = 0
+        changed = False
+        while i < 10 and not changed:
+            i += 1
+            time.sleep(5)  # give some time to pacemaker to react
+            new_crm_master = self.get_crm_master()
+
+            if (new_crm_master and new_crm_master != old_crm_master):
+                logging.info(
+                    "New crm_master unit detected"
+                    " on {}".format(new_crm_master)
+                )
+                changed = True
```

Comment:
You could use an 'else' clause to your while loop, remove the 'changed' variable and change this line to a break. But I guess its a question of preference.